sdk/core/azure-core/src/http/curl/curl_session_private.hpp

```diff
@@ -238,16 +238,20 @@ namespace Azure { namespace Core { namespace Http {
      * inner buffer. When a libcurl stream tries to read part of the body, this field will help to
      * decide how much data to take from the inner buffer before pulling more data from network.
      *
+     * @note The index will never be greater than `int32_t` because the size of the inner buffer has
+     * type `int32_t`. See m_innerBufferSize.
      */
-    int64_t m_bodyStartInBuffer = -1;
+    int32_t m_bodyStartInBuffer = -1;
 
     /**
      * @brief Control field to handle the number of bytes containing relevant data within the
      * internal buffer. This is because internal buffer can be set to be size N but after writing
      * from wire into it, it can be holding less then N bytes.
      *
+     * @note Field will never be greater than `int32_t` because the size of the inner buffer has
+     * type `int32_t`. See m_innerBufferSize.
      */
-    int64_t m_innerBufferSize = _detail::DefaultLibcurlReaderSize;
+    int32_t m_innerBufferSize = _detail::DefaultLibcurlReaderSize;
```

Comment:
`size_t`?

Reply:
The internal buffer might even be smaller.  We use it just to pull the status line and headers from the payload in chunks of 1024 bytes.